src/rust/engine/src/externs/fs.rs

```diff
@@ -116,6 +116,18 @@ py_class!(pub class PySnapshot |py| {
       Self::create_instance(py, Snapshot::empty())
     }
 
+    @classmethod def _create_for_testing(
+      _cls,
+      py_digest: PyDigest,
+      files: Vec<String>,
+      dirs: Vec<String>,
+    ) -> PyResult<Self> {
+      let snapshot = unsafe {
```

Comment:
Similar point made elsewhere about use of `unsafe`. Needs a comment explaining safety.

Reply:
Normally I'd agree, but this seems fine imo. This function is a private one only used by FFI and both names make clear this is all for the sake of testing, rather than prod use.